openstates/md/bills.py

```diff
@@ -165,10 +170,132 @@ def parse_bill_votes(self, doc, bill):
+        lines = text.splitlines()
+
+        if 'senate' in vote_url:
+            chamber = 'upper'
+        else:
+            chamber = 'lower'
+
+        date_string = lines[0].split('Date:')[1].strip()
+        date = datetime.datetime.strptime(date_string, "%b %d, %Y %I:%M (%p)")
+
+        pageIndex = -1
+        for index, line in enumerate(lines):
+            if 'Yeas' in line and 'Nays' in line:
+                pageIndex = index
+                break
+
+        yes_count = -1
+        no_count = -1
+        other_count = 0
+
+        if pageIndex > 0:
+
+            counts = re.split(r'\s{2,}', lines[pageIndex].strip())
+
+            for count in counts:
+                number, string = count.split(' ', 1)
+                number = int(number)
+
+                if string == 'Yeas':
+                    yes_count = number
+                elif string == 'Nays':
+                    no_count = number
+                else:
+                    other_count += number
+
+        passed = yes_count > no_count
+
+        motion = re.split(r'\s{2,}', lines[pageIndex-3].strip())[0]
+        if motion == '':
+            motion = "No motion given"
+
+        vote = VoteEvent(
+            bill=bill,
+            chamber=chamber,
+            start_date=date.strftime('%Y-%m-%d'),
+            motion_text=motion,
+            classification='passage',
+            result='pass' if passed else 'fail',
+        )
+
+        vote.pupa_id = vote_url  # contains sequence number
+
+        vote.set_count('yes', yes_count)
+        vote.set_count('no', no_count)
+        vote.set_count('other', other_count)
+        pageIndex = pageIndex + 2
+
+        # Keywords for identifying where names are located in the pdf
+        showStoppers = ['Voting Nay', 'Not Voting',
+                        'COPY', 'Excused', 'indicates vote change']
+        voteTypes = ['yes', 'no', 'other']
+        voteIndex = 0
+
+        # For matching number of names extracted with vote counts(extracted independently)
+        totalvotes = 0
+        yes_names_count = 0
+        no_names_count = 0
+
+        while pageIndex < len(lines):
+
+            currentLine = lines[pageIndex].strip()
+
+            if not currentLine or 'Voting Yea' in currentLine:
+                pageIndex += 1
+                continue
+
+            if any(showStopper in currentLine for showStopper in showStoppers):
+                pageIndex += 1
+                voteIndex = (voteIndex + 1) if (voteIndex < 2) else 2
+                continue
+
+            names = re.split(r'\s{2,}', currentLine)
+
+            if voteIndex == 0:
+                yes_names_count += len(names)
+            elif voteIndex == 1:
+                no_names_count += len(names)
+
+            totalvotes += len(names)
+            for name in names:
+                vote.vote(voteTypes[voteIndex], name)
+            pageIndex += 1
+
+        if yes_count != yes_names_count:
```

Comment:
I'd argue that if any of these counts don't match, then we've failed our scrape, and should either 1) throw an error or 2) not record individuals' votes (ie, set the `Vote.yes`, `Vote.no`, etc lists to empty/null).

Reply:
Yes, here actually I was throwing error only locally, thought we may want to keep running it. Doing it right away.